design/service-library.md

```diff
@@ -0,0 +1,187 @@
+
+## NodeJS end-to-end using the Service library
+
+### Concept
+
+This example is utilizing both the core and the service library. That means that as a developer, we want the full fluent API, allowing me discover APIs via the intellisence. Because the service library leverages the core library, we also want to highlight that we can leverage the core components directly (like the `.api()` method).
+
+### Install the Microsoft Graph JavaScript Service Library
+
+There will be two versions of the service library for the Graph JavaScript API Client. The default one will bring the representation of the v1.0 of Microsoft Graph. The second one will bring the representation of the beta of Microsoft Graph. This will be the only difference between these two libraries, as all the handcrafted code (delivering authentication, middlewares, tasks, etc.) will happen in the core library.
+
+The Microsoft Graph JavaScript Service Library includes the following packages:
+
+- `@microsoft/microsoft-graph-client` ([npm](https://www.npmjs.com/package/@microsoft/microsoft-graph-client)) - The service library for making fluent calls to Microsoft Graph v1.0.
+- `@microsoft/microsoft-graph-client-beta` ([npm](https://www.npmjs.com/package/@microsoft/microsoft-graph-client-beta)) - The service library for making fluent calls to Microsoft Graph beta.
+
+You can use [npm](https://www.npmjs.com) to install the Microsoft Graph JavaScript Service Library:
+
+```shell
+npm install @microsoft/microsoft-graph-client --save
+npm install @microsoft/kiota-authentication-azure --save
+npm install @azure/identity --save
+```
+
+### Importing the right functionalities from the Graph JavaScript Service Library
+
+```typescript
+import { Client } from "@microsoft/microsoft-graph-client";
+import { User, Message, BodyType } from "@microsoft/microsoft-graph-client/models";
+import { AzureIdentityAuthenticationProvider } from "@microsoft/kiota-authentication-azure";
+import { DeviceCodeCredential } from "@azure/identity";
+```
+
+### Setting up the app to work with the Graph JavaScript Service Library
+
+```typescript
+const deviceCodeCredentials = new DeviceCodeCredential({
+	tenantId: "b61f9af1-d6cf-4cc0-a6f6-befb38bc00ed",
+	clientId: "bde251a6-0ef9-42a8-a40b-9ad9bb594b2c",
+});
+
+const scopes = ["User.Read", "Mail.Send"];
+
+const graphClient = Client.init({
+	authenticationTokenProvider: new AzureIdentityAuthenticationProvider(deviceCodeCredentials, scopes),
+});
+```
+
+### Calling Microsoft Graph via the Graph JavaScript Service Library
+
+The core value of the Microsoft Graph JavaScript Service Library is the availability of the chaned request builder pattern (fluent API). This provides developers with typechecking and discoverability when building on Microsoft Graph. This would highlight the full spectrum of capabilities on Microsoft Graph directly in the API Client.
```

Comment:
```suggestion
The core value of the Microsoft Graph JavaScript Service Library is the availability of the chained request builder pattern (fluent API). This provides developers with typechecking and discoverability when building on Microsoft Graph. This would highlight the full spectrum of capabilities on Microsoft Graph directly in the API Client.
```